test/e2e/cross_os/test_job_submissions.py

```diff
@@ -312,6 +313,498 @@ def is_expected_session_action_canceled(sessions: List[Dict[str, Any]]) -> bool:
+                "name": "TestEnvJobFail",
+                "jobEnvironments": [
+                    {
+                        "name": successful_environment_name,
+                        "script": {
+                            "actions": {
+                                "onEnter": ({"command": "whoami"}),
+                                "onExit": ({"command": "whoami"}),
+                            },
+                        },
+                    },
+                    {
+                        "name": unsuccessful_environment_name,
+                        "script": {
+                            "actions": {
+                                "onEnter": ({"command": "nonexistentcommand"}),
+                                "onExit": ({"command": "nonexistentcommand"}),
+                            },
+                        },
+                    },
+                ],
+                "steps": [
+                    {
+                        "name": "Step0",
+                        "hostRequirements": {
+                            "attributes": [
+                                {
+                                    "name": "attr.worker.os.family",
+                                    "allOf": [os.environ["OPERATING_SYSTEM"]],
+                                }
+                            ]
+                        },
+                        "script": {
+                            "actions": {
+                                "onRun": ({"command": "whoami"}),
+                            }
+                        },
+                    },
+                ],
+            },
+        )
+        # THEN
+
+        # Wait until the job is completed
+        job.wait_until_complete(client=deadline_client)
+
+        sessions = deadline_client.list_sessions(
+            farmId=job.farm.id, queueId=job.queue.id, jobId=job.id
+        ).get("sessions")
+
+        found_successful_env_enter = False
+        found_unsuccessful_env_enter = False
+        found_unsuccessful_env_exit = False
+        found_successful_env_exit = False
+
+        # Find that the both the unsuccessful and successful environment ran, with envExit and envEnter for each.
+        for session in sessions:
+
+            session_actions = deadline_client.list_session_actions(
+                farmId=job.farm.id,
+                queueId=job.queue.id,
+                jobId=job.id,
+                sessionId=session["sessionId"],
+            ).get("sessionActions")
+            logging.info(f"Session actions: {session_actions}")
+            for session_action in session_actions:
+                definition = session_action["definition"]
+                if "envEnter" in definition:
+                    if successful_environment_name in definition["envEnter"]["environmentId"]:
+                        assert session_action["status"] == "SUCCEEDED"
+                        found_successful_env_enter = True
```

Comment:
nit, since we're asserting the intersection of all of these values, I think we can set these "found_(un)successful..." variables to the expression.

ie, `found_successful_env_enter = session_action["status"] == "SUCCEEDED"`

Reply:
Makes sense, will revise